File: src/easy/0100-same-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */
var isSameTree = function(p, q) {
  var isSameNode = function(pNode, qNode) {
    if (pNode === null && qNode === null) return true;
    if (pNode === null || qNode === null) return false;
    if (pNode.val !== qNode.val) return false;
    return isSameNode(pNode.left, qNode.left) && isSameNode(pNode.right, qNode.right);
  }
  return isSameNode(p, q);
};

File: src/easy/0100-same-tree.cpp
/* 思路：只需同时遍历两棵树，判断每个位置上的值是否相同即可。可以采取先序，中序，后序均可达到目的。*/
class Solution {
public:
	bool isSameTree(TreeNode* p, TreeNode* q) {
		if (p == NULL && q == NULL) return true;
		if (p == NULL || q == NULL) return false;
		if (p->val != q->val) return false;
		return isSameTree(p->left, q->left) && isSameTree(p->right, q->right);

	}
};
